src/main/java/com/umc/withme/service/MemberService.java

```diff
@@ -19,6 +22,19 @@ public class MemberService {
      * @return 중복인 경우 - true, 중복이 아닌 경우 - false
      */
     public boolean checkNicknameDuplication(String nickname) {
-        return memeberRepository.existsByNickname(nickname);
+        return memberRepository.existsByNickname(nickname);
+    }
+
+    /**
+     * 특정 닉네임을 가진 회원 정보를 컨트롤러에게 반환한다.
+     *
+     * @param nickname
+     * @return MemberDto에 회원 정보를 담아 반환
+     * @throws NicknameNotFoundException 닉네임이 존재하지 않을 경우
+     */
+    public MemberDto getMemberInfo(String nickname) {
+        Member findMember = memberRepository.findByNickname(nickname)
+                                    .orElseThrow(() -> new NicknameNotFoundException());
+        return MemberDto.from(findMember);
```

Comment:
Comment가 많아서 좀 죄송하기도 한데,,, 마지막으로 하나만 더 달겠습니다. 하하 😅;;
`Optional`은 mapping을 지원하므로 다음과 같이 하면 변수 선언 없이 필요한 dto를 return 할 수 있어 코드가 조금 더 깔끔해질 것 같습니다!

```suggestion
    return memberRepository.findByNickname(nickname)
            .map(MemberDto::from)
            .orElseThrow(NicknameNotFoundException::new);
```

Reply:
아닙니다! 제가 놓친 부분에 대해 피드백 주셔서 감사합니당😀 반영 후 푸쉬했습니당